File: set/src/action/PlayerAction.java
package src.action;

import src.proto.AllProtos;

/**
 * @author dylanblecher
 * The type of action that can be made by a player
 * (as opposed to SHOW_SET and DRAW_3 which can only be requested through REQUEST_SHOW_SET for example.)
 * PlayerActions include all other actions.
 * Proto-backed according to Google's Protobuf
 */
public class PlayerAction extends Action {
    public PlayerAction(ActionType type, int playerID) {
        super(AllProtos.Action.newBuilder().setType(type.proto).setPlayerID(playerID).build());
    }

    public PlayerAction(AllProtos.Action action) {
        super(action);
    }

    public int getPlayerID() {
        return proto.getPlayerID();
    }
}

File: set/src/client/PlayerAction.py
class PlayerAction:
	def __init__(self, actionType, playerID):
		self.actionType = actionType
		self.playerID = playerID

	def getActionType(self):
		return self.actionType

	def getPlayerID(self):
		return self.playerID

class PlayerActionSelectSet(PlayerAction):
	def __init__(self, actionType, playerID, cardPositions):
		super().__init__(actionType, playerID)
		self.cardPositions = cardPositions

	def getCardPositions(self):
		return self.cardPositions